### DISCUSSION TITLE
How to do headers on all except 1 route / 1 set of routes?

### DISCUSSION THREAD

**Discussion by vedantroy** [remix-run/remix#2849] [Q&A] [answered]
Let's say I have the following setup:
- foo.com/posts/:postId
- foo.com/questions/:questionId
- foo.com/admin/:adminId

And, I don't want to show the header & footer on the admin routes. Is there a good way to do this?
By default, I'd add the header & footer to the root route, but this would cause them to show on the admin route as well.
I'm wondering if there's a clean solution for only showing it on certain child routes but not others.

I could just repeat the header & footer code in both the posts & questions routes, but this seems well ... repetitive.

**Comment by fnimick** [comment, marked as answer]
I think you can use pathless routing for this. IIRC can have the following setup:
```
- __index/posts/...
- __index/questions/...
- __index/index.tsx (for your root route, wrapped in the common header)
- __index.tsx (for your header)
- admin/... (for admin routes, not wrapped in the header layout)`
```

If you want your `/` to not have the common header, rename `__index` to some other name, and delete `__index/index.tsx`.
